presto-spi/src/main/java/com/facebook/presto/spi/plan/AggregationNode.java

```diff
@@ -210,7 +210,12 @@ public PlanNode replaceChildren(List<PlanNode> newChildren)
 
     public boolean isStreamable()
     {
-        return !preGroupedVariables.isEmpty() && groupingSets.getGroupingSetCount() == 1 && groupingSets.getGlobalGroupingSets().isEmpty();
+        return !preGroupedVariables.isEmpty() && groupingSets.getGroupingSetCount() == 1 && groupingSets.getGlobalGroupingSets().isEmpty() && preGroupedVariables.size() == groupingSets.groupingKeys.size();
```

Comment:
This line is kind of long. I would split it up across multiple lines for readability